mirrord/agent/src/runtime/error.rs

```diff
@@ -0,0 +1,39 @@
+use thiserror::Error;
+
+/// Errors that can occur when fetching target container info.
+#[derive(Error, Debug)]
+pub(crate) enum ContainerRuntimeError {
+    #[error("failed to get target container info: {} [{}]", .error, .runtime)]
+    GetInfoError { runtime: String, error: String },
+    #[error("unknown container runtime `{0}`")]
+    UnknownRuntimeName(String),
+}
+
+impl ContainerRuntimeError {
+    pub(crate) fn crio<E: ToString>(error: E) -> Self {
+        Self::GetInfoError {
+            runtime: "cri-o".into(),
+            error: error.to_string(),
+        }
+    }
+
+    pub(crate) fn docker<E: ToString>(error: E) -> Self {
+        Self::GetInfoError {
+            runtime: "docker".into(),
+            error: error.to_string(),
+        }
+    }
+
+    pub(crate) fn containerd<E: ToString>(error: E) -> Self {
+        Self::GetInfoError {
+            runtime: "containerd".into(),
+            error: error.to_string(),
+        }
+    }
+
+    pub(crate) fn unknown_runtime<N: ToString>(name: N) -> Self {
+        Self::UnknownRuntimeName(name.to_string())
+    }
+}
+
+pub(crate) type Result<T, E = ContainerRuntimeError> = core::result::Result<T, E>;
```

Comment:
This pattern kinda always annoys me, when the alias is named just `Result`, and I want to check something out by writing a quick function that returns a `Result<(), SomeOtherError>`.

Could you rename it to `ContainerResult` (or something else you find appropriate)?